pkg/grpc/databroker/client.go

```diff
@@ -17,3 +26,90 @@ type staticClientGetter struct {
 func (w staticClientGetter) GetDataBrokerServiceClient() DataBrokerServiceClient {
 	return w.client
 }
+
+type databrokerClientKey struct{}
+
+func WithDatabrokerClient(ctx context.Context, client dataBrokerServiceClient) context.Context {
+	return context.WithValue(ctx, databrokerClientKey{}, client)
+}
+
+func GetDatabrokerClient(ctx context.Context) DataBrokerServiceClient {
+	c, ok := ctx.Value(databrokerClientKey{}).(DataBrokerServiceClient)
+	if !ok {
+		c = &nilClient{
+			defaultErr: status.Error(codes.Unavailable, "not yet initialized"),
+		}
+	}
+	return c
+}
+
```

Comment:
@wasaga what do you think about injecting the databroker client into a context, that way we don't have to couple the interface and overrides to include the databroker client directly in signatures?

Just a thought experiment on my part, but maybe this will work fine